File: src/webscrapingDatabase/database/webscrape_class.js
const mysql = require("mysql")
const axios = require("axios")
const cheerio = require("cheerio")

class Object{
    constructor(element, attribute, class_name){
        this.element = element
        //variable for specific html element
        this.attribute = attribute
        //variable for elements attributes
        this.class_name = class_name
        //variable for elements class
    }
}

function finding_inputting(url, object, cals) {
    axios(url)
        .then(response => {
            const html_script = response.data
            const values = []
            const calVals = []
            const honey = cheerio.load(html_script)
            honey(object.class_name, html_script).each(function(){
                //function created to search through entire script
                const text = honey(this).text()
                const att = honey(this).find(object.element).attr(object.attribute)


                values.push({
                    text
                })
            })

            honey(cals.class_name, html_script).each(function(){
                const text = honey(this).text()
                calVals.push({
                    text
                })
            })

            //const valuesTwo = values
            var con = mysql.createConnection({
                //creation of connection to MySQL database
                    host: "allergen-db1.cutjc5tgzcff.us-east-2.rds.amazonaws.com",
                    user: "admin",
                    password: "Bismarck66!",
                    database: "Foods"
                })

                con.connect(function(err){
                    if (err){
                        throw err;
                    }
                    console.log("connected")

                    var databaseValues = [[]]
                    //2d variable
                    for(let i = 0; i < values.length; i++){
                        databaseValues[[i]] =
                            [i, values[i].text, calVals[i].text, "yes", "yes", "no", "Pfeiffer"]
                    }

                    con.query("DELETE FROM Food", function(err, result){
                        if(err){
                            throw err
                        }
                        console.log("rows affected " + result.affectedRows)
                    })

                    con.query("INSERT INTO Food (Food_ID, Food_Name, Calories, Peanut_Free, Dairy_Free, Gluten_Free, Location) VALUES ?",
                     [databaseValues], function(err, result){
                    //querying of webscraped data
                        if(err) {
                            throw err;
                        }
                        console.log("records inserted " + result.affectedRows);
                    })

                })
                for(let i = 0; i < values.length; i++){
                    console.log(values[i])
                }


        }).catch(err => console.log(err))
}


objectOne = new Object('a', 'href', '.get-nutritioncalculator')
//foods

cals = new Object('a', 'href', '.get-nutrition')
//calories of foods

finding_inputting('https://menus.sodexomyway.com/BiteMenu/Menu?menuId=14740&locationId=10395001&whereami=https://simpsondining-preview.sodexomyway.com/dining-near-me/pfieffer-dining-hall',
objectOne, cals)

File: src/webscrapingDatabase/database/webscrape_class.py
from bs4 import BeautifulSoup
import requests
import mysql.connector

#creation of mysql connection
db = mysql.connector.connect(
    host="allergen-db1.cutjc5tgzcff.us-east-2.rds.amazonaws.com",
    user='admin',
    password='Bismarck66!',
    database='Foods'
)

mycursor = db.cursor()

mycursor.execute("DELETE FROM Food")

#retreival of website data
url = "https://menus.sodexomyway.com/BiteMenu/Menu?menuId=14740&locationId=10395001&whereami=https://simpsondining-preview.sodexomyway.com/dining-near-me/pfieffer-dining-hall"
result = requests.get(url)
doc = BeautifulSoup(result.text, "html.parser")

database_values = [[]]
#arrays for specific items
food = []
calorie = []
peanuts = []
dairy = []
gluten = []

"""Beginning of Webscraping script"""
#html parsers for specific items
foods = doc.find_all("a", attrs={"class":"get-nutritioncalculator"})
calories = doc.find_all("a", attrs={"class":"get-nutrition"})
allergens = doc.find_all("img", attrs={"class":"icon-allergen"})

for i in range(0, len(foods)):
    calorie.append(calories[i].text.strip())
    if(allergens[i]['alt'] == 'contains peanuts'):
        peanuts.append("contains peanuts")
    else:
        peanuts.append('peanut free')

    if(allergens[i]['alt'] == 'contains milk'):
       dairy.append('contains milk')
    else:
        dairy.append("dairy free")

    if(allergens[i]['alt'] == 'contains gluten'):
        gluten.append('contains gluten')
        #gluten.append("contains gluten")
    else:
        gluten.append("gluten free")

    """end of webscraping script"""


    """Beginning of database pushing script"""
    if foods[i].text == 'Peripherals':
        #database_values.append([foods[i].text, '0cal' 'peanut free', 'dairy free', 'gluten free'])
        mycursor.execute("INSERT INTO Food (Food_ID, Food_Name, Calories, Peanut_Free, Dairy_Free, Gluten_Free) VALUES (%s, %s, %s, %s, %s, %s)",
        (i, foods[i].text, "0cal", "peanut free", "dairy free", "gluten free"))
    else:
        #database_values.append([foods[i].text, calorie[i], peanuts[i], dairy[i], gluten[i]])
        mycursor.execute("INSERT INTO Food (Food_ID, Food_Name, Calories, Peanut_Free, Dairy_Free, Gluten_Free) VALUES (%s, %s, %s, %s, %s, %s)",
        (i, foods[i].text, calorie[i], peanuts[i], dairy[i], gluten[i]))

db.commit()
"""end of database pushing script"""
